compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -717,8 +730,7 @@ object CheckUnused:
 
       /** A function is overriden. Either has `override flags` or parent has a matching member (type and name) */
       private def isOverriden(using Context): Boolean =
-        sym.is(Flags.Override) ||
-          (sym.exists && sym.owner.thisType.parents.exists(p => sym.matchingMember(p).exists))
+        sym.is(Flags.Override) || lookForInheritedDecl(sym).isDefined
```

Comment:
Is there a test covering a case for which this check had to be modified?

Reply:
That modification has been necessary to overcome a new non handled exception.

That exception appeared when putting the `CheckUnused` and `CheckShadowing` phases in the same `MegaPhase`.
In facts when doing the same traversing for both phases, it seems that some Trees becomes empty values along with their symbols.
That work around solved all the test covering cases that appears in the `CommunityBuildTestA` and `CommunityBuildTestC`. One of them that I remembered exactly is the **cask** project.

Reply:
That's so strange... I'd rather we knew about the cause, but we should be all right if tests and community builds pass.

Reply:
Yes, I never knew the exact cause honestly. During debugging, I simply saw that my way to look after inherited symbols with the same name (in `CheckShadowing`'s `lookForShadowedField()` method) was more robust for handling the possible exceptions when reaching an empty `Trees`.
Then for the global functionality, I indeed relied on the existent test cases.